[docs] live improvements (extended version) (#489)
canonical/rockcraft: docs/index.rst

   .. grid-item-card:: :ref:`Tutorials <tutorials>`

>       **Get started** - become familiar with Rockcraft by containerising different kinds of software as rocks.

Reviewer: ```suggestion
      **Get started** - become familiar with Rockcraft by containerising different software applications as rocks.
```

   .. grid-item-card:: :ref:`How-to guides <how-to>`

>       **Step-by-step guides** - learn key operations about slices and rocks,
>       ranging from creating and cutting slices to migrating and publishing rocks.

Reviewer: ```suggestion
      **Step-by-step guides** - learn key operations,
      ranging from creating and cutting slices to migrating and publishing rocks.
```

.. grid:: 1 1 2 2
   :reverse:

   .. grid-item-card:: :ref:`Reference <reference>`

      **Technical information** - understand how to use every field in
      ``rockcraft.yaml``.

   .. grid-item-card:: :ref:`Explanation <explanation>`

      **Discussion and clarification** - explore Rockcraft lifecycle and how a

>       rock image gets packed under the hood.

Reviewer: ```suggestion
      rock gets packed under the hood.
```
